safesocket/pipe_windows.go

```diff
@@ -39,5 +50,100 @@ func listen(path string) (net.Listener, error) {
 	if err != nil {
 		return nil, fmt.Errorf("namedpipe.Listen: %w", err)
 	}
-	return lc, nil
+	return &listenerWrap{Listener: lc}, nil
+}
+
+// ConnWithToken is an implementation of net.Conn that permits retrieval of
+// the Windows access token associated with the connection's client.
+type ConnWithToken struct {
+	net.Conn
+	token windows.Token
+}
+
+func (conn *ConnWithToken) handle() windows.Handle {
+	fd := conn.Conn.(interface {
+		Fd() uintptr
+	})
```

Comment:
so this just panics if you use the wrong embedded Conn?

use a comma-ok type assertion to not panic?

pull this interface out into an unexported named type with some comments about who implements it?